Sources/Tracking/CustomerIOImplementation+ScreenViews.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+#if canImport(UIKit)
+import UIKit
+
+public extension CustomerIOImplementation {
+    func enableAutoScreenviewTracking() {
+        let selector1 = #selector(UIViewController.viewDidAppear(_:))
+        let selector2 = #selector(CustomerIOImplementation._swizzled_UIKit_viewDidAppear(_:))
+        guard let originalMethod = class_getInstanceMethod(UIViewController.self, selector1) else {
+            return
+        }
+        guard let swizzleMethod = class_getInstanceMethod(CustomerIOImplementation.self, selector2) else {
+            return
+        }
+        method_exchangeImplementations(originalMethod, swizzleMethod)
+    }
+
+    @objc dynamic func _swizzled_UIKit_viewDidAppear(_ animated: Bool) {
+        _swizzled_UIKit_viewDidAppear(animated)
+
+        guard let delegate = UIApplication.shared.delegate else {
+            return
+        }
+
+        guard let window = delegate.window else {
+            return
+        }
+
+        var viewController = window!.rootViewController
+        if viewController is UINavigationController {
+            viewController = (viewController as! UINavigationController).visibleViewController
+        }
+
+        var name = String(describing: type(of: viewController)).replacingOccurrences(of: "ViewController", with: "")
+
+        if name.isEmpty || name == "" {
+            if viewController?.title != nil {
+                name = viewController?.title ?? "Unknown"
+            }
+            if name.isEmpty || name == "" {
+                // Could not infer screen name
+                name = "Unknown"
+            }
+        }
+
+        screen(name: name, data: autoScreenViewBody()) { _ in
+            // TODO: global error handling of result here
+        }
+    }
+}
+
+#else
+public extension CustomerIOImplementation {
+    func enableAutoScreenviewTracking() {
+        // TODO: log warning that tracking is not available
```

Comment:
Our Swift linting tool marks `TODO` as a warning. I prefer we stick with `XXX: ` to make the linter happy. I like this lint behavior because you can still use TODO as personal reminders to yourself before your work is done but it's not the best for longer term future work like in this case, logging.

Reply:
👍